gnovm/cmd/gno/fix.go

```diff
@@ -143,6 +154,164 @@ func gnoFixParseGnomod(dir string) (mod *gnomod.File, isDotMod bool) {
 	return
 }
 
+func txtarsFromArgs(target string) ([]string, error) {
+	var paths []string
+
+	info, err := os.Stat(target)
+	if err != nil {
+		return []string{}, fmt.Errorf("gno: invalid file %q: %w", target, err)
+	}
+
+	if !info.IsDir() {
+		if isTxtarFile(fs.FileInfoToDirEntry(info)) {
+			paths = append(paths, cleanPath(target))
+		}
+		return paths, nil
+	}
+
+	files, err := os.ReadDir(target)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
+		if isTxtarFile(f) {
+			path := filepath.Join(target, f.Name())
+			paths = append(paths, cleanPath(path))
+		}
+	}
+
+	return paths, nil
+}
+
+func isTxtarFile(f fs.DirEntry) bool {
+	return strings.HasSuffix(f.Name(), ".txtar") && !f.IsDir()
+}
+
+func (c *fixCmd) processFixTxtar(file string) error {
+	archive, err := txtar.ParseFile(file)
+	if err != nil {
+		return err
+	}
+	// group files by folder to handle error gnomod versions.
+	var filesByDir = map[string][]txtar.File{}
+	for _, f := range archive.Files {
+		dir := filepath.Dir(f.Name)
+		filesByDir[dir] = append(filesByDir[dir], f)
+	}
+	for _, files := range filesByDir {
+		res, err := c.processFixTxtarDir(files)
+		if err != nil {
+			return err
+		}
+		// This keep orders to limit diff noise.
+		// NOTE: this is due to processing txtars by folders breaking the order.
+		for _, mf := range res {
+			for i := range archive.Files {
+				if archive.Files[i].Name == mf.Name {
+					archive.Files[i].Data = mf.Data
+				}
+			}
+		}
+	}
+	if !c.diff {
+		archive := txtar.Format(archive)
+		info, err := os.Stat(file)
+		if err != nil {
+			return fmt.Errorf("stat txtar file: %w", err)
+		}
+		err = os.WriteFile(file, archive, info.Mode())
+		if err != nil {
+			return fmt.Errorf("error writing txtar file: %w", err)
+		}
+	}
+	return nil
+}
+
+func (c *fixCmd) processFixTxtarDir(files []txtar.File) ([]txtar.File, error) {
+	var gm *gnomod.File
+	var res []txtar.File
+	for _, f := range files {
+		if f.Name == "gnomod.toml" {
+			gm, _ = gnomod.ParseBytes("gnomod.toml", f.Data)
+			break
+		}
+	}
+	for _, f := range files {
```

Comment:
Do you think there's a way to join most of the code of this loop with that of `processFix`